examples/jsm/utils/BufferGeometryUtils.js

```diff
@@ -199,7 +202,22 @@ function mergeBufferGeometries( geometries, useGroups = false ) {
 		mergedGeometry.userData.mergedUserData = mergedGeometry.userData.mergedUserData || [];
 		mergedGeometry.userData.mergedUserData.push( geometry.userData );
 
-		if ( useGroups ) {
+		if ( useGroups && geometry.groups.length > 0 ) {
+		
+			// Keep existing groups
+			
+			for (let group of geometry.groups ) {
+				
+				let offsetMaterialIndex = groupIndex + group.materialIndex
+				mergedGeometry.addGroup( offset, group.count,  offsetMaterialIndex);
+				offset += group.count;
+				nextGroupIndex = Math.max( nextGroupIndex, offsetMaterialIndex );
+				
+			}
+			
+			++ nextGroupIndex;
```

Comment:
nit: we'll want to fix the code styling to follow three.js style. That includes spaces inside of parentheses and newlines inside curly braces.

Reply:
I think this should be correct now. (Although sometimes it breaks my brain and I don't even see the three.js formatting errors)